chef/cookbooks/provisioner/templates/suse/crowbar_register.erb

```diff
@@ -260,9 +266,17 @@ post_state() {
 # -----------------------
 
 MAC=$(cat /sys/class/net/$BOOTDEV/address)
-HOSTNAME="d${MAC//:/-}.${DOMAIN}"
+if test $KEEP_EXISTING_HOSTNAME -ne 1; then
+  HOSTNAME="d${MAC//:/-}.${DOMAIN}"
+  echo "$HOSTNAME" > /etc/HOSTNAME
+else
+  HOSTNAME=$(cat /etc/HOSTNAME)
+  if ! echo "$HOSTNAME" | grep -q -o "$DOMAIN$"; then
```

Comment:
Strictly speaking, you actually need to also check that there's no dot once you remove the `.$DOMAIN` suffix from `$HOSTNAME`.

(I'd also change the grep to use `.$DOMAIN$`)